File: toy/next-board/server/src/main/java/me/kzv/server/domain/Comment.java
package me.kzv.server.domain;

import lombok.Builder;
import lombok.Getter;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;
import java.util.Optional;
import java.util.stream.IntStream;
import java.util.stream.Stream;

@Getter
@Entity
public class Comment extends BaseEntity{

    @Id
    @GeneratedValue
    @Column(name = "comment_id")
    private Long id;
    @Column(columnDefinition = "text")
    private String body;
    private String username;

    private Long postId;


    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "post_id")
    private Post post;


    @OneToMany(mappedBy = "comment")
    private List<Vote> votes = new ArrayList<>();

    protected Comment() {
    }

    @Builder
    public Comment(Long id, String body, String username, Long postId) {
        this.id = id;
        this.body = body;
        this.username = username;
        this.postId = postId;
    }

    public void changeVote(User user) {
    }
}

File: hello-kotlin/kotlin-board/src/main/kotlin/me/kzv/board/entity/Comment.kt
package me.kzv.board.entity

import javax.persistence.*

@Entity
class Comment private constructor(id: Long?, content: String, writer: String, userId: Long, postId: Long) : BaseTimeEntity() {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "comment_id")
    var id: Long? = id
        private set

    var content: String = content
        private set

    var writer: String = writer
        private set

    var userId: Long = userId
        private set

    var postId: Long = postId
        private set

    constructor(content: String, writer: String, userId: Long, postId: Long) : this(null, content, writer, userId, postId)
}